### ISSUE TITLE
[Feature Request] Apex support for mixed precision training

### ISSUE THREAD

**Issue by nijkah** [open-mmlab/mmcv#1719]
**Describe the feature**

**Motivation**
Many recent research works are based on apex for mixed precision training.

So I want to add this feature to mmcv with the name of class EpochBasedRunnerApex.
I am thinking implementation of just wrapping EpochBasedRunner as in [Swin-Transformer Implementation](https://github.com/SwinTransformer/Swin-Transformer-Object-Detection/blob/master/mmcv_custom/runner/epoch_based_runner.py).

Is it good idea to mmcv? Then I will create the PR.

**Related resources**
https://github.com/SwinTransformer/Swin-Transformer-Object-Detection
https://github.com/facebookresearch/ConvNeXt
https://github.com/microsoft/unilm/tree/master/beit/semantic_segmentation

**Comment by HAOCHENYE** [comment]
Thanks for your advice, but what is the advantage of apex over torch.cuda.amp? mmcv implements mixed precision training via Fp16OptimizerHook.

**Comment by nijkah** [comment]
@HAOCHENYE 
[From this link](https://discuss.pytorch.org/t/torch-cuda-amp-vs-nvidia-apex/74994/7), it says Apex is a first implementation of amp, and deprecated.
So, it doesn't need to support that.
I think I have to check to check why these works uses apex for amp.
